### ISSUE TITLE
Alleviate subtle gotcha regarding '#head' packages

### ISSUE THREAD

**Issue by dom96** [nim-lang/nimble#332]
I just installed ``nimble@0.8.4`` and was wondering why its code was not being imported in ``picknim`` (picknim reuses Nimble's code for pure unadulterated awesomness). The reason was that I also had ``nimble@#head`` installed. At that point ``nimble@#head`` was actually older than ``nimble@0.8.4``.

Should Nimble have given me a warning during the installation of ``nimble@0.8.4``? Something like "hey, nimble@#head is now older than the one you just installed, wanna update #head?"

**Comment by Araq** [comment]
I think it's simply wrong to have multiple directories with $version appended, instead Nimble should control git repositories and use ``git checkout`` to get a specific version before builds. Nimble needs enforce some git tagging scheme for this to work though.

**Comment by dom96** [comment]
Not every package is inside a git repo. This will be even more true in the future.

**Comment by Araq** [comment]
Well it can also be in a hg repo.

> This will be even more true in the future.

No, it won't, version control is everywhere. And nimble can simply require a hg/git repo.

**Comment by dom96** [comment]
> No, it won't, version control is everywhere. And nimble can simply require a hg/git repo.

Yes, it will. Unless you want our packages site to host these git repos.

**Comment by Araq** [comment]
I don't understand this. Right now nimble uses git/hg to download packages.

**Comment by dom96** [comment]
Yes, and this has a major problem. Packages can go missing at any second. We need a package site where we can upload packages to so that we can ensure they remain available.

**Comment by Araq** [comment]
The site should just fork or clone the originals. Why throw away information?

**Comment by dom96** [comment]
So I think we can do that for the package site.

But something else to consider is speed. Won't having to ``checkout`` constantly be incredibly slow?
